build/commands/lib/checkNodeNpmVersion.js

```diff
@@ -0,0 +1,65 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+const assert = require('assert')
+const Log = require('./logging')
+
+// Define the required versions.
+const requiredNodeVersionRange = ['20.0.0', '*']
+const requiredNpmVersionRange = ['10.0.0', '*']
+
+// Get the actual versions.
+const nodeVersion = process.versions.node
+const npmVersion = process.env.npm_config_npm_version
+
+// Check Node.js version.
+checkVersionInRange('node', nodeVersion, requiredNodeVersionRange)
+
+// Check npm version if it's defined. It can be undefined if Yarn is used.
+if (npmVersion !== undefined) {
+  checkVersionInRange('npm', npmVersion, requiredNpmVersionRange)
+}
+
+// Check if a version is within a min-max range (excluding max unless max is
+// '*').
+function checkVersionInRange(type, version, range) {
+  const [min, max] = range
+  const versionParts = version.split('.').map(Number)
+  assert(versionParts.length === 3)
```

Comment:
Could it be something like `20.10.5-something` in some linux distros?

Reply:
> Could it be something like `20.10.5-something` in some linux distros?

Added `parseInt` to ignore possible non-digits in the version.